acceptance/cli/cli.go

```diff
@@ -714,15 +714,39 @@ func logExecution(ctx context.Context) {
 	fmt.Print(output.String())
 }
 
-func matchSnapshot(ctx context.Context) error {
+func matchSnapshot(ctx context.Context, formatJson bool) error {
 	status, err := ecStatusFrom(ctx)
 	if err != nil {
 		return err
 	}
 
-	stdout := snaps.MatchSnapshot(ctx, "stdout", status.stdout.String(), status.vars)
-	stderr := snaps.MatchSnapshot(ctx, "stderr", status.stderr.String(), status.vars)
+	var stdout, stderr error
+	
+	
+	//formatJson := false // Wont allow me put it in the function call
```

Comment:
How about setting `formatJSON` based on whether or not it can be decoded as JSON? Something like this:

```diff
diff --git a/acceptance/cli/cli.go b/acceptance/cli/cli.go
index cff84ca..1617cf8 100644
--- a/acceptance/cli/cli.go
+++ b/acceptance/cli/cli.go
@@ -714,39 +714,32 @@ func logExecution(ctx context.Context) {
 	fmt.Print(output.String())
 }
 
-func matchSnapshot(ctx context.Context, formatJson bool) error {
+func matchSnapshot(ctx context.Context) error {
 	status, err := ecStatusFrom(ctx)
 	if err != nil {
 		return err
 	}
 
 	var stdout, stderr error
-	
-	
-	//formatJson := false // Wont allow me put it in the function call
-	if(formatJson){
-		formatStdout := status.stdout
-		var parsedOutput map[string]interface{}
-		err := json.Unmarshal([]byte(formatStdout.Bytes()), &parsedOutput)
-		if err != nil {
-			return err
-		}
 
-		prettyOutput, err := json.MarshalIndent(parsedOutput, "", "    ")
+	formatJSON := true
+	var jsonOutput json.RawMessage
+	if err := json.Unmarshal(status.stdout.Bytes(), &jsonOutput); err != nil {
+		formatJSON = false
+	}
+
+	if formatJSON {
+		prettyOutput, err := json.MarshalIndent(jsonOutput, "", "    ")
 		if err != nil {
 			return err
-			
 		}
-		
 		stdout = snaps.MatchSnapshot(ctx, "stdout", string(prettyOutput), status.vars)
 		stderr = snaps.MatchSnapshot(ctx, "stderr", status.stderr.String(), status.vars)
-
-	}else{
-
+	} else {
 		stdout = snaps.MatchSnapshot(ctx, "stdout", status.stdout.String(), status.vars)
 		stderr = snaps.MatchSnapshot(ctx, "stderr", status.stderr.String(), status.vars)
 	}
-	
+
 	if stdout == nil && stderr == nil {
 		return nil
 	}

```

Also, I think unmarshaling to `json.RawMessage` makes this function a little more flexible.